microdnf/commands/makecache/makecache.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "makecache.hpp"
+
+#include "microdnf/context.hpp"
+
+#include <iostream>
+
+
+namespace microdnf {
+
+
+using namespace libdnf::cli;
+
+
+MakeCacheCommand::MakeCacheCommand(Command & parent) : Command(parent, "makecache") {
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_short_description("Generate the metadada cache");
+}
+
+
+void MakeCacheCommand::run() {
+    auto & ctx = static_cast<Context &>(get_session());
+
+    libdnf::repo::RepoQuery enabled_repos_query(ctx.base);
+    enabled_repos_query.filter_enabled(true);
+    if (enabled_repos_query.empty()) {
+        std::cout << "There are no enabled repositories." << std::endl;
```

Comment:
In dnf4 we also add list of directories where .repo are searched. Do you want to add it?

Reply:
I don't know.
Example of using DNF4:
```
#dnf --disablerepo=* makecache
There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
```
But I have the repositories enabled in the specified configuration files. Only the settings are overwritten from the command line (or also the plugin).
In addition, repositories can also be defined in the main configuration (dnf.conf). The message in DNF4 seems misleading to me.

Reply:
OK. It adds information. It's a little misleading, but it can still help sometimes.  I made a change. The same message is now displayed as in DNF4.
Maybe we'll come up with a less misleading sentence in the future.